packages/apps/shopify-app-remix/src/server/authenticate/admin/strategies/merchant-custom-app.ts

```diff
@@ -0,0 +1,62 @@
+import {AppConfig, AppConfigArg} from 'src/server/config-types';
+import {Session, Shopify, ShopifyRestResources} from '@shopify/shopify-api';
+import {BasicParams} from 'src/server/types';
+import {
+  ApiConfigWithFutureFlags,
+  ApiFutureFlags,
+} from 'src/server/future/flags';
+import {HandleAdminClientError} from 'src/server/clients';
+
+import {handleClientErrorFactory} from '../helpers';
+
+import {AuthorizationStrategy, OnErrorOptions, SessionContext} from './types';
+
+export class MerchantCustomAuth<Config extends AppConfigArg>
+  implements AuthorizationStrategy
+{
+  protected api: Shopify<
+    ApiConfigWithFutureFlags<Config['future']>,
+    ShopifyRestResources,
+    ApiFutureFlags<Config['future']>
+  >;
+
+  protected config: AppConfig;
+  protected logger: Shopify['logger'];
+
+  public constructor({api, config, logger}: BasicParams<Config['future']>) {
+    this.api = api;
+    this.config = config;
+    this.logger = logger;
+  }
+
+  public async respondToOAuthRequests(_request: Request): Promise<void> {
+    this.logger.debug('Skipping OAuth request for merchant custom app');
+  }
+
+  public async authenticate(
+    _request: Request,
+    sessionContext: SessionContext,
+  ): Promise<Session | never> {
+    const {shop} = sessionContext;
+
+    this.logger.debug(
+      'Building session from configured access token for merchant custom app',
+    );
+    const session = this.api.session.customAppSession(shop);
+
+    return session;
+  }
+
+  public handleClientError(request: Request): HandleAdminClientError {
+    // const {api, config, logger} = this;
+    return handleClientErrorFactory({
+      request,
+      onError: async ({error}: OnErrorOptions) => {
+        if (error.response.code === 401) {
+          // What should we do if we receive a 401 error?
```

Comment:
What should we do here if we receive a 401?

Reply:
Agreed, these requests should explode to remind the developer to update their token. Maybe even mention that in the error message?

Reply:
<img width="506" alt="image" src="https://github.com/Shopify/shopify-app-js/assets/23265671/9911d5d8-4890-4c44-a97b-be30c95fee08">

<img width="1270" alt="image" src="https://github.com/Shopify/shopify-app-js/assets/23265671/39fe6cc2-38e0-486c-90d7-a0d7a64ed1c0">

Throws an error and prints a log with a link to the token rotation docs.